src/cocotb/types/_logic_array.py

```diff
@@ -252,6 +252,19 @@ def __init__(
                 )
             self._value_as_int = value
             self._range = range
+        elif isinstance(value, LogicArray):
+            array = value._value_as_array
+            self._value_as_array = list(array) if array is not None else None
+            self._value_as_int = value._value_as_int
+            self._value_as_str = value._value_as_str
+            if range is None:
+                self._range = value._range
+            else:
+                if len(range) != len(value):
+                    raise ValueError(
+                        f"{value!r} will not fit in a LogicArray with bounds: {range!r}"
```

Comment:
```suggestion
                        f"{value!r} will not fit exactly in a LogicArray with bounds: {range!r}"
```
Maybe? Not what this PR is about, but while we are here...
There is another such message in line 251.